### PR TITLE
Extracts CLI parsing from Main, writes a test for edge case

### PR DESCRIPTION
**Pull request by colindean**
Apparently, scopt doesn't provide a way to escape commas in
a comma-separated key-value list option.

https://github.com/scopt/scopt/issues/111

This test documents that and enables us to test fixes for it.

### CODE REVIEW

**Comment by colindean** [comment]
I'll revisit this after #65 is merged.

File: src/main/scala/com/target/data_validator/CliOptionParser.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package com.target.data_validator
+
+import scopt.OptionParser
+
+case class CmdLineOptions(
```
Comment (dbseraf1) [inline src/main/scala/com/target/data_validator/CliOptionParser.scala:5]: I assume you want to be backward compatible to reduce the number of changes in this PR, but this would be better named "CliOptions" to be consistent with "CliOptionParser"

Comment (colindean) [reply src/main/scala/com/target/data_validator/CliOptionParser.scala:5]: I think it's pretty safe to effect that sensible change in this PR. Done.

**Comment by colindean** [comment]
I'm not a fan of how ScalaStyle+IntelliJ indented those args in ccadbaa but I'll allow it.

File: src/test/scala/com/target/data_validator/CliOptionParserSpec.scala [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.target.data_validator
+
+import org.scalatest.{FunSpec, Matchers}
+
+class CliOptionParserSpec extends FunSpec with Matchers {
+
+  describe("CliOptionParser") {
+    describe("parsing") {
+      it("does not handle var option values with commas") {
+        val args = Array("--vars", "keyA=value1,value2,keyB=value3")
```
Comment (c-horn) [inline src/test/scala/com/target/data_validator/CliOptionParserSpec.scala:10]: Could we just use a different delimiter inside the list? is there a use case for list variables?

**Review by c-horn — APPROVED** [review]

File: src/test/scala/com/target/data_validator/CliOptionParserSpec.scala (hunk 2 above)
Comment (colindean) [reply src/test/scala/com/target/data_validator/CliOptionParserSpec.scala:10]: It's unfortunately inflexible. See https://github.com/scopt/scopt/issues/111 as well as the commit message on one of the commits in this PR.
